### ISSUE TITLE
no-dupe-class-members and ClassProperty

### ISSUE THREAD

**Issue by Turbo87** [eslint/eslint#13059]
**What rule do you want to change?**

no-dupe-class-members

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

new default behavior

**Please provide some example code that this change will affect:**

```js
class A { foo = 42; get foo() {} }
```

**What does the rule currently do for this code?**

it does not warn about the `ClassProperty`

**What will the rule do after it's changed?**

it warns that there are multiple `foo` properties on the class

**Are you willing to submit a pull request to implement this change?**

yes

the only issue is that `ClassProperty` comes from `babel-eslint`, and I'm not sure if the "fix" should be done here or in https://github.com/babel/eslint-plugin-babel 🤔

**Comment by mdjermanovic** [comment]
Hi @Turbo87, thanks for the issue! 

> the only issue is that `ClassProperty` comes from `babel-eslint`, and I'm not sure if the "fix" should be done here or in https://github.com/babel/eslint-plugin-babel

As you noted, core rules don't support `ClassProperty` since it isn't defined in the [ESTree specification](https://github.com/estree/estree), as this syntax is still in stage 3.

Looks like the solution at the moment would be indeed to implement this rule in `eslint-plugin-babel`.

**Comment by Turbo87** [comment]
okay, thanks for the feedback :)
